File: 0703-kth-largest-element-in-a-stream/0703-kth-largest-element-in-a-stream.cpp
class KthLargest {
public:
    priority_queue<int,vector<int> , greater<int>>pq; 
    int K;
    KthLargest(int k, vector<int>& nums) {
        int n = nums.size();
        K=k;
        // put all the elements in pq
        for(int i=0;i<n;i++){
            pq.push(nums[i]);
            if(pq.size()>k){
                pq.pop() ;// top element (jo smallest hoga) uskon uda do
            }
        }
    }
    int add(int val) {
        pq.push(val);
        if(pq.size()>K){
            pq.pop();
        }
        return pq.top();
    }
};

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest* obj = new KthLargest(k, nums);
 * int param_1 = obj->add(val);
 */

File: 0703-kth-largest-element-in-a-stream/0703-kth-largest-element-in-a-stream.py
class KthLargest:

    def __init__(self, k: int, nums: List[int]):
        self.k = k
        self.heap = []
        for num in nums:
            heapq.heappush(self.heap , num) # heap me nums ka elelmnets daal do
            if(len(self.heap) > k):
                heapq.heappop(self.heap) # agar heap me k se jyada element hai to pop kr do .... heap me bss k elements hi rakhna hai 

    def add(self, val: int) -> int:
        heapq.heappush(self.heap, val) # jb new element aa raha ho to heap me dsal do 
        if(len(self.heap) > self.k):
            heapq.heappop(self.heap)
        return self.heap[0]

        


# Your KthLargest object will be instantiated and called as such:
# obj = KthLargest(k, nums)
# param_1 = obj.add(val)